### DISCUSSION TITLE
Capture Mouse Event in a DataGridTextColumn

### DISCUSSION THREAD

**Discussion by drei97** [AvaloniaUI/Avalonia#13646] [Q&A]
Hi,

I need a context menu (or Flyout) for a special column in a DataGrid. The right mouse button should bring up the menu only if the mouse pointer is within a defined column.

How to do it?

The PointerPressedEvent for the DataGrid doesn't trigger for a column, only for areas which are not a part of a column or a row, and DataGridTextColumn doesn't provide mouse events.

I'am running 11.0.5

Joachim

**Comment by timunie** [comment]
Maybe it can be added for DataGridCell via Behaviors? 

I personally use a TemplateColumn and just added the ContextFlyout in the CellTemplate
